common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -253,26 +330,51 @@ public ICacheRecord save(@NonNull AccountRecord accountRecord,
                     familyId
             );
 
-            final String clientId = cacheRecord.getAccessToken().getClientId();
-            final String target = cacheRecord.getAccessToken().getTarget();
-            final String environment = cacheRecord.getAccessToken().getEnvironment();
+            return loadAggregatedAccountData(authScheme, cacheRecord);
+        }
+    }
 
-            // Now get the cache we just saved to....
-            final MsalOAuth2TokenCache cache = getTokenCacheForClient(
-                    clientId,
-                    environment,
-                    mCallingProcessUid
+    public synchronized List<ICacheRecord> saveAndLoadAggregatedAccountData(
+            final @NonNull AccountRecord accountRecord,
+            final @NonNull IdTokenRecord idTokenRecord,
+            final @NonNull AccessTokenRecord accessTokenRecord,
+            final @NonNull RefreshTokenRecord refreshTokenRecord,
+            final @Nullable String familyId,
+            final @NonNull AbstractAuthenticationScheme authScheme) throws ClientException {
+        synchronized (this) {
+            final ICacheRecord cacheRecord = save(
+                    accountRecord,
+                    idTokenRecord,
+                    accessTokenRecord,
+                    refreshTokenRecord,
+                    familyId
             );
 
-            return (List<ICacheRecord>) cache.loadWithAggregatedAccountData(
-                    clientId,
-                    target,
-                    cacheRecord.getAccount(),
-                    authScheme
-            );
+            return loadAggregatedAccountData(authScheme, cacheRecord);
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private List<ICacheRecord> loadAggregatedAccountData(final @NonNull AbstractAuthenticationScheme authScheme,
+                                                         final @NonNull ICacheRecord cacheRecord) {
+        final String clientId = cacheRecord.getAccessToken().getClientId();
+        final String target = cacheRecord.getAccessToken().getTarget();
+        final String environment = cacheRecord.getAccessToken().getEnvironment();
+
+        final MsalOAuth2TokenCache cache = getTokenCacheForClient(
+                clientId,
+                environment,
+                mCallingProcessUid
+        );
+
+        return (List<ICacheRecord>) cache.loadWithAggregatedAccountData(
```

Comment:
Not immediately clear to me why this result needs to be casted, doesn't this API always return this type?

Reply:
This is automatically generated by the extract function in Android Studio. Will remove 😅